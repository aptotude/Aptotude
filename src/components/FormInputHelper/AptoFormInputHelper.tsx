import React from 'react';
import { InputProps } from 'semantic-ui-react';
import { AptoInput } from '../Input';
import { AptoFormControlError } from '../FormControlError';
import { AptoForm } from '../Form/AptoForm';

export interface AptoInputOnChangeData extends InputProps {
  value: string;
}

export interface AptoFormInputHelperProps {
  required?: boolean;
  label: string;
  name: string;
  value: any;
  error?: string;
  change: (
    event: React.ChangeEvent<HTMLInputElement>,
    data: AptoInputOnChangeData
  ) => void;
  length?: number;
}

export const AptoFormInputHelper = (
  props: AptoFormInputHelperProps
): JSX.Element => {
  const { required, label, name, value, error, change, length } = props;
  const id = `input-${name}`;
  return (
    <AptoForm.Field required={required}>
      <label htmlFor={id}>{label}</label>
      <AptoInput
        maxLength={length || 255}
        required={required}
        value={value || ''}
        error={error ? true : undefined}
        onChange={change}
        name={name}
        id={id}
      />
      <AptoFormControlError error={error} />
    </AptoForm.Field>
  );
};
