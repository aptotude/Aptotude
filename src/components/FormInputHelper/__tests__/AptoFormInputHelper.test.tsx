import React from 'react';
import { render, cleanup } from 'react-testing-library';
import { AptoFormInputHelper } from '../AptoFormInputHelper';

afterEach(cleanup);

describe('AptoFormInputHelper', () => {
  it('should render ', () => {
    const props = {
      required: true,
      label: 'Label',
      name: 'field',
      value: 16,
      error: undefined,
      change: () => {
        // wee
      },
      length: 24
    };
    const { container } = render(<AptoFormInputHelper {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render error state', () => {
    const props = {
      label: 'Label',
      name: 'field',
      value: 16,
      error: 'Broken Stuff',
      change: () => {
        // wee
      }
    };
    const { container } = render(<AptoFormInputHelper {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
