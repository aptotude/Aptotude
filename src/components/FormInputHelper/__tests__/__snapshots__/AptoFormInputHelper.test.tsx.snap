// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AptoFormInputHelper should render  1`] = `
<div
  class="required field"
>
  <label
    for="input-field"
  >
    Label
  </label>
  <div
    class="ui input"
  >
    <input
      id="input-field"
      maxlength="24"
      name="field"
      required=""
      type="text"
      value="16"
    />
  </div>
</div>
`;

exports[`AptoFormInputHelper should render error state 1`] = `
<div
  class="field"
>
  <label
    for="input-field"
  >
    Label
  </label>
  <div
    class="ui error input"
  >
    <input
      id="input-field"
      maxlength="255"
      name="field"
      type="text"
      value="16"
    />
  </div>
  <div
    class="AptoFormControlError"
  >
    Broken Stuff
  </div>
</div>
`;
